player/lua/stats.lua

```diff
@@ -77,6 +78,8 @@ local o = {
     no_ass_b0 = "\027[0m",
     no_ass_it1 = "\027[3m",
     no_ass_it0 = "\027[0m",
+
+    ["help-keys"] = "no",  -- print the list on startup and quit mpv
```

Comment:
Is there a reason why this uses `-` instead of `_` as the other options do? There is something about this at line 1144 but I don't get it.
`print the list` -> what list? Is it the same output otherwise printed on screen?

Reply:
> Is there a reason why this uses - instead of _ as the other options do?

Other options use "-", and for users it's way more convenient to use "-" instead of "_" (it annoys me to no end when I need to use "_" for words separation in script opts).

Only script options usually use "_", and only because it's more convenient to use at the code itself (can be used as `opts.foo_bar` instead of the more awkward `opts["foo-bar"]`.

> There is something about this at line 1144 but I don't get it.

Not sure I understand. Please be more specific.

> `print the list` -> what list? Is it the same output otherwise printed on screen?

Yes. Same output which can be browsed interactively at the video window or at the terminal OSD. It can be printed with the same form to the terminal like a help page.

Reply:
> `print the list` -> what list?

The first comment has some collapsed content. Expand it to see how it looks at the terminal.

Reply:
While I understand your sentiment regarding dash/underscore options, it's a personal preference and we don't even know if your subjective one is representative for other users. However, this makes the stats script options inconsistent with a mix of dash and underscore options. I would really prefer to avoid that.
 
(Note: personally I have no preference other than consistency)

Reply:
> However, this makes the stats script options inconsistent with a mix of dash and underscore options. I would really prefer to avoid that.

To me using mixed dash/underscore is inconsistent on its own, and using dashes-only is consistent on its own and also consistent with general mpv options.

I definitely understand where you're comming from, but I still strongly prefer dashes-only.

Maybe others could share their preference?

Reply:
I think `keys_help` is more consistent because all `script-opts` use _. If i were to type this on the terminal from memory months from now, I would use _ and expect it to work.
I also find that using different symbols for separating the script and its options and for the options themselves helps to differentiate them visually.

Reply:
> I think `keys_help` is more consistent because all `script-opts` use _

No, they use both. For instance `--script-opts=ytdl_hook-ytdl_path=whatever`, and It's a damn pain to remember which are `-` and which are `_`. Personally I almost never get the dashes/underscores correct on first attempt.

Some of the underscores are sort of forced, because mpv convers script-names to use underscores (that's the `ytdl_hook` part). Then there's always a dash, and then the actual options are usually with underscore - for no good reason other than it's more convinient to write at the source code `o.foo_bar` instead of `o["foo-bar"]`.

So it ends up that if the script name is one word and the option name is one word, then it always uses only dashes. This is very common, for instance `--script-opts=osc-visibility=always` or `--script-opts=stats-font=whatever`.

However, script name and/or option names which are more than one word are separated with underscore.

Of the builtin scripts, only `ytdl_hook` is two words, and many of the option-names are one word, so in practice many of script opts of most builtin scripts are dashes-only.

Personally, I really dislike the forced mixed dash/underscores. However:
- If my opinion is a minority then I'll cave.
- On such case, I'd prefer to find one word so that you guys feel it's consistent with other script-opts. Maybe `--script-opts=stats-keys=help` or some such?

Reply:
> If my opinion is a minority then I'll cave.

I don't think if this should be about opinions, for me this is about consistency. You demonstrated how the current system as a whole is inconsistent and I give you that but at least the current stats script-options are consistent and use only underscores. I would really prefer not to include an unnecessary and additional inconsistency here now.

Now, changing _all_ of stats' script-options from underscores to dashes is a possibility. As said before, I have no personal preference here, so I'd be fine to change all options to dashes if that helps with consistency. However, this would render all existing `script-opts/stats.conf` invalid and, therefore, should imo only be done when there's a strong majority among users in favor of this change.

Reply:
> I don't think if this should be about opinions, for me this is about consistency.

Yeah, I see your point, but I think differently _on this specific case_. I'm fine with doing what most think is best. No hard feeelings here.

> Now, changing _all_ of stats' script-options from underscores to dashes is a possibility.

Nah, too painful for users and for anyone who might try that. IMHO the only way to support consistent dashes is to automate it so that both would work for script-opts - like they are for mpv command-names (dashes and underscores are interchangable). This way no script needs to be modified, and nothing breaks (or very little breaks where they conclist), and everyone is happy.

However, that's not for this PR.

So, you don't want dash-separated option names, and I don't want underscores.

How about you help me find one word so that it doesn't need to use underscores, and we're both happy?

What about `--script-opts=stats-helpkeys=yes` ? or `stats-printkeys=yes`? (combined words are very common at the `osc.lua` for instance) or `--script-opts=stats-keys=help`? Or something else which you have in mind?

Reply:
Also, if you look at the commit message which added this option, you'll see that I fully acknowledged the awkwardness of the invocation, though not due to the dashes (this actually makes it better than underscores), but rather because it's basically a help page which is triggered via script-opts because that's what scripts can use.

I mentioned that in the future we can try to streamline the invocation, and @guidocella suggested something like `--input-bindings` or `input-keylist` (similar to the existing `--list-protocols` and `--input-cmdlist`). That would definitely be better than the approach of this PR.

However, it also doesn't contradict. We can have something now via script-opts to make it useful, and when we find the time to add more streamlined invocation, we'll do that too.

Reply:
> What about ...

How about `--script-opts=stats-bindlist=yes` in similar naming as the existing `--input-cmdlist` and `--input-keylist` ?

I'm going to assume this is acceptable unless there's an explicit objection (at which case do suggest an altrenative using one word).